tests/kv2.rs

```diff
@@ -47,6 +47,42 @@ fn test() {
     });
 }
 
+#[test]
+fn test_kv2_url_encoding() {
+    let test = common::new_test();
+    test.run(|instance| async move {
+        let server: VaultServer = instance.server();
+        let client = server.client();
+
+        debug!("setting up kv2 auth engine");
+        let path = "path/to/secret engine";
+        let name = "path/to/some secret/password name with whitespace";
+        let secret = TestSecret {
+            key: "mykey".to_string(),
+            password: "supersecret".to_string(),
+        };
+        let endpoint = SecretEndpoint {
+            path: path.to_string(),
+            name: name.to_string(),
+            secret,
+        };
+
+        // Mount the KV2 engine
+        server.mount_secret(&client, path, "kv-v2").await.unwrap();
+
+        // Create a test secret
+        create(&client, &endpoint).await.unwrap();
+
+        let secrets = kv2::list(&client, path, "path/to/some secret/").await.unwrap();
+        assert_eq!(secrets.len(), 1);
+        assert_eq!(secrets.get(0).unwrap(), "password name with whitespace");
+
+        let res: Result<TestSecret, _> = kv2::read(&client, path, name).await;
+        assert!(res.is_ok());
+        assert_eq!(res.unwrap().key, endpoint.secret.key);
+    });
+}
+
```

Comment:
This causes on my system an error because it tries to run two vault servers with the same port.
Please write an async function and pass the vault server to it.